### PR TITLE
Add docs about which licenses are ok to use

### PR DESCRIPTION
**Pull request by chrismwendt**
From https://sourcegraph.slack.com/archives/C01GJFE26BX/p1633126727289600?thread_ts=1632329123.208100&cid=C01GJFE26BX

### CODE REVIEW

File: content/engineering/index.md [hunk 1]
```diff
@@ -29,6 +29,7 @@
 - [Career development](career-development/index.md)
   - [Career development framework](career-development/framework.md)
   - [Talent review process](career-development/talent-review-process.md)
+- [Permitted licenses](licenses.md)
```
Comment (emidoots) [inline content/engineering/index.md]:
```suggestion
- [Allowed open source licenses](licenses.md)
```

**Review by emidoots — APPROVED** [review]
I pushed some changes. Will explain in thread: https://sourcegraph.slack.com/archives/C01GJFE26BX/p1632329123208100

File: content/engineering/licenses.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+- AGPL
+- GPL (ok for internal use)
+- LGPL (ok for internal use)
+- CCDL (ok for internal use)
+- Eclipse Public License (ok for internal use)
+- Mozilla Public License (ok for internal use)
+- Anything else not on the permissive license list
+
+If it's for internal use only and not for distribution to users/customers, then tell the legal team and the licenses above will likely be approved.
```
Comment (bobheadxi) [inline content/engineering/licenses.md:32]: @chrismwendt  Note that this is semi-enforced here: https://github.com/sourcegraph/sourcegraph/blob/main/doc/dependency_decisions.yml we should make sure these lists stay in sync

Comment (bobheadxi) [reply content/engineering/licenses.md:32]: https://docs.sourcegraph.com/dev/background-information/continuous_integration#third-party-licenses

**Pull request merged**